Add autoquant in LLM benchmarking CI

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141910

Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Reduce num_layers to enable autoquant in CI

Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Reduce num_layers to enable autoquant in CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Reduce num_layers to enable autoquant in CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Reduce num_layers to enable autoquant in CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Reduce num_layers to enable autoquant in CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Reduce num_layers to enable autoquant in CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Reduce num_layers to enable autoquant in CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Reduce num_layers to enable autoquant in CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

huydhn reviewed (approved):
LGTM! The model shows up on [the dashboard](https://hud.pytorch.org/benchmark/llms?startTime=Fri%2C%2029%20Nov%202024%2018%3A38%3A57%20GMT&stopTime=Fri%2C%2006%20Dec%202024%2018%3A38%3A57%20GMT&granularity=hour&lBranch=ciflow/inductor-micro-benchmark/141910&lCommit=0b7b6152e629780c61c1636842952421a970756c&rBranch=ciflow/inductor-micro-benchmark/141910&rCommit=e99647c09350e59339e7c4e011d6907c794a01c2&repoName=pytorch%2Fpytorch&modelName=All%20Models&dtypeName=autoquant&deviceName=All%20Devices)

yanboliang reviewed (dismissed):
sorry I think we should only reduce num of layers for a particular model, because the perf number is correlated with the num of layers. I'd like to suggest to define a subclass for the ```run_llama2_7b_autoquant``` model, otherwise, we have to adjust all of these per numbers.

jerryzh168 committed:
Update on "Reduce num_layers to enable autoquant in CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@yanboliang sounds good. I reverted the n_layer changes, seems to be fine

jerryzh168 commented:
@pytorchbot merge

jerryzh168 committed:
Update on "Add autoquant in LLM benchmarking CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Add autoquant in LLM benchmarking CI"


Summary:
original autoquant will require A100-80G to run in CI, but we don't have that right now, so reducing num_layer so that we can
track autoquant in dashboard

Test Plan:
dashboard
https://hud.pytorch.org/benchmark/llms?repoName=pytorch%2Fpytorch

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
Some issues with this right now
running llama-autoquant standalone:
```
Compilation time: 82.14 seconds
Average tokens/sec: 115.20 tokens/sec
Average bandwidth achieved: 1522.33 GB/s
Memory used: 14.99 GB
```


running both int8 llama and llama-autoquant together:
```
Compilation time: 283.86 seconds
Average tokens/sec: 110.52 tokens/sec
Average bandwidth achieved: 1460.47 GB/s
Memory used: 15.01 GB
Loading model Llama-2-7b-chat-hf
Time to load model: 0.14 seconds
Compilation time: 283.28 seconds
Average tokens/sec: 87.44 tokens/sec
Average bandwidth achieved: 1155.46 GB/s
Memory used: 17.52 GB
```

Repro: python benchmark.py

switch between two experiments: https://github.com/pytorch/pytorch/pull/141910/files#diff-5412ff53aeb6bc81cc81dcfb32204983950a098d7fb89fd28bc7876dd862c633R321-R331
